ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -184,53 +393,191 @@ def test_from_dlpack(
     )
 
 
-# full()
+# full
+@given(
+    shape=helpers.get_shape(
+        allow_none=False,
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+    ),
+    fill_value=st.integers(0, 5) | st.floats(0.0, 5.0),
+    dtype=st.sampled_from(ivy_np.valid_numeric_dtypes),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="full"),
+    container=st.booleans(),
+)
+def test_full(
+    shape,
+    fill_value,
+    dtype,
+    with_out,
+    device,
+    num_positional_args,
+    container,
+    fw,
+):
+
+    if type(fill_value) == float and dtype in ivy_np.valid_int_dtypes:
```

Comment:
we should omit this case during the data generation phase, rather than in the function body itself. You might need to create a short helper function for this.